internal/pkg/util/names.go

```diff
@@ -34,3 +44,42 @@ func Contains(a []string, b string) bool {
 	}
 	return false
 }
+
+// LengthName creates a string of maximum defined length.
+func lengthName(maxLen int, hashPrefix, format string, a ...interface{}) (string, error) {
+	friendlyName := fmt.Sprintf(format, a...)
+	if len(friendlyName) < maxLen {
+		return friendlyName, nil
+	}
+
+	// If that's too long, just hash the name. It's not very user friendly, but whatever
+	//
+	// We can suppress the gosec warning about sha1 here because we don't use sha1 for crypto
+	// purposes, but only as a string shortener
+	// nolint:gosec
+	hasher := sha1.New()
```

Comment:
Completely understand the rational here. I would probably use sha256 though, given the project's focus in security we should refrain from taking dependencies on algorithmns or libraries that are known to be weak/vulnerable.

This may also be a problem for users in high compliance sectors, as this is not NIST compliant.

Reply:
Thank you for the review. I understand where you're coming from, but the problem is the 64 characters length that labels are restricted to in k8s. Consider this (it's python but I hope it's readable nonetheless):
```
In [5]: hashlib.sha1('foo'.encode()).hexdigest()
Out[5]: '0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33'

In [6]: hashlib.sha256('foo'.encode()).hexdigest()
Out[6]: '2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae'

In [7]: len(hashlib.sha1('foo'.encode()).hexdigest())
Out[7]: 40

In [8]: len(hashlib.sha256('foo'.encode()).hexdigest())
Out[8]: 64
```
so sha256 doesn't already fit on its own:
```
kubectl label sp log-enricher-trace hash=2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae
error: invalid label value: "hash=2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae": must be no more than 63 characters
```

One thing we could do (but..it's kindof a hack..) is to trim the length a bit, e.g. use an equivalent of git's abbrev and fall back to using the full hash without prefix if there is a collision, which frankly is quite unlikely. And even if there was a collision, it wouldn't be that much of a deal as the hash is not used for any security critical functionality, just a link between two objects.

Reply:
If you are cool with using sha256, trimming the tail end of it is a common approach which slighty increases chance of collision but that isn't a problem on that case - and less likely to occur than sha1 anyway. 

Otherwise, would you be opposed to simply restricting the size of profile name and using that instead? 

In both the approaches above, any concerns around name collision through profiles with the same name deployed in different namespaces?

Now if this can be non-deterministic, we could also just append the hex of some random values at the tail end using `rand.Read(buf)`.

Reply:
> If you are cool with using sha256, trimming the tail end of it is a common approach which slighty increases chance of collision but that isn't a problem on that case - and less likely to occur than sha1 anyway.

How much is reasonable to trim? As little as possible to fill the 63 characters?

> Otherwise, would you be opposed to simply restricting the size of profile name and using that instead?

I would not be opposed to that. The only catch is that we use sometimes derive security profile resources from other resources (e.g. we derive from recording name), so we need to be careful and restrict them all or we might be unable to generate the profiles later.

> In both the approaches above, any concerns around name collision through profiles with the same name deployed in different namespaces?

Both objects are namespaced and live in the same namespace, so I think using the namehash is no different than using the plain name. Or did you have some other issue in mind?

> Now if this can be non-deterministic, we could also just append the hex of some random values at the tail end using rand.Read(buf).

I honestly prefer deterministic at least to make testing easier.

Reply:
ok, let me start by trimming the hash and in parallel look at limiting the name. We can compare the two approaches then (and see test results!)

Reply:
> How much is reasonable to trim? As little as possible to fill the 63 characters?

Yep, as little as possible would be ideal.

> I would not be opposed to that. The only catch is that we use sometimes derive security profile resources from other resources (e.g. we derive from recording name), so we need to be careful and restrict them all or we might be unable to generate the profiles later.

Very good point. That's definitely something we want to get right and aligned sooner rather than later.

> Both objects are namespaced and live in the same namespace, so I think using the namehash is no different than using the plain name. Or did you have some other issue in mind?

That's right - same issue in both cases. I was thinking this could be a problem when we deploy SPO to watch multiple namespaces for Security Profiles, which I think we still support - but not completely sure.